miio/yeelight.py

```diff
@@ -109,30 +212,66 @@ def __init__(self, *args, **kwargs):
         default_output=format_output(
             "",
             "Name: {result.name}\n"
-            "Power: {result.is_on}\n"
-            "Brightness: {result.brightness}\n"
-            "Color mode: {result.color_mode}\n"
-            "RGB: {result.rgb}\n"
-            "HSV: {result.hsv}\n"
-            "Temperature: {result.color_temp}\n"
             "Developer mode: {result.developer_mode}\n"
             "Update default on change: {result.save_state_on_change}\n"
+            "Delay in minute before off: {result.delay_off}\n"
+            "Music mode: {result.music_mode}\n"
+            "Light\n"
+            "   Power: {result.is_on}\n"
+            "   Brightness: {result.brightness}\n"
+            "   Color mode: {result.color_mode}\n"
+            "   RGB: {result.rgb}\n"
+            "   HSV: {result.hsv}\n"
+            "   Temperature: {result.color_temp}\n"
+            "   Color flowing mode: {result.color_flowing}\n"
+            "   Color flowing parameters: {result.color_flow_params}\n"
+            "Moonlight\n"
+            "   Is in mode: {result.moonlight_mode}\n"
+            "   Moonlight mode brightness: {result.moonlight_mode_brightness}\n"
+            "Background light\n"
+            "   Power: {result.is_bg_on}\n"
+            "   Brightness: {result.bg_brightness}\n"
+            "   Color mode: {result.bg_color_mode}\n"
+            "   RGB: {result.bg_rgb}\n"
+            "   HSV: {result.bg_hsv}\n"
+            "   Temperature: {result.bg_color_temp}\n"
+            "   Color flowing mode: {result.bg_color_flowing}\n"
+            "   Color flowing parameters: {result.bg_color_flow_params}\n"
             "\n",
         )
     )
     def status(self) -> YeelightStatus:
         """Retrieve properties."""
         properties = [
+            # general properties
+            "name",
+            "lan_ctrl",
+            "save_state",
+            "delayoff",
+            "music_on",
+            # light properties
             "power",
             "bright",
-            "ct",
+            "color_mode",
             "rgb",
             "hue",
             "sat",
-            "color_mode",
-            "name",
-            "lan_ctrl",
-            "save_state",
+            "ct",
+            "flowing",
+            "flow_params",
+            # moonlight properties
+            "active_mode",
+            "nl_br",
+            # background light properties
+            "bg_power",
+            "bg_bright",
+            "bg_lmode",
+            "bg_rgb",
+            "bg_hue",
+            "bg_sat",
+            "bg_ct",
+            "bg_flowing",
+            "bg_flow_params",
```

Comment:
I haven't yet checked this PR out with my lights (which do not support background lights), but it may be needed to do separate queries depending on the light. I'll try to find some time in the near future to check that out and I'll report back.

Reply:
I already checked it. And its work but looks not cool:
```
Name: 
Developer mode: True
Update default on change: True
Delay in minute before off: 0
Music mode: None
Light
   Power: False
   Brightness: 100
   Color mode: 2
   RGB: None
   HSV: None
   Temperature: 5200
   Color flowing mode: False
   Color flowing parameters: None
Moonlight
   Is in mode: False
   Moonlight mode brightness: 0
Background light
   Power: None
   Brightness: None
   Color mode: None
   RGB: None
   HSV: None
   Temperature: None
   Color flowing mode: None
   Color flowing parameters: None
```
Where I can look on separated subdevices. (My background is C++ and python is additional language for me, i don't really know how @command(default output...) works.)

Reply:
About the command output, I wouldn't worry about it that much, the cli tool is there just for completeness as the main use case for this lib is to use it as a library :-)

Changing that will require some restructuring / work, so I think the best thing to do (for the time being) is to document which properties are supported by which models.

I am wondering if we could find a way to share the "database" of device features with python-yeelight as they already have collected information that could be useful in the future? See https://gitlab.com/stavros/python-yeelight/-/blob/master/yeelight/main.py#L29 - ping @skorokithakis

Reply:
Hmm, yes, that's a good idea! Maybe we could make a separate common library, to avoid adding either as a dependency of the other.

Reply:
What can I help with this? As I understand we must wait answer from @skorokithakis? 
As for me. I sink good idea is moving shared data to external lib.

Reply:
Isn't the data all we both need? We could agree on a common format and make a new library that's simply this data.

Reply:
@skorokithakis  for my opinion your data format is great. But I thinked about simple getter like  
`get_known_models` and maybe something like `get_specs_for_model(str: model_name)` because this 2 function looks like general for any libs. 
But it may be only the data. 
I only start to work with this and not sure what data Is important for me in future. If you know any other general datas that you have I`ll be very heppy if you shared it.

@rytilahti What do you think?

Reply:
That makes sense, but it sounds like those getters will be aliases to `MODEL_DICT.keys()` and `MODEL_DICT[model_name]`, which isn't really a hassle to write. Then again, I'm not against writing a simple class either.

Reply:
I'm going to unfollow this PR because I got 17 emails, please open an issue if you decide something on the library matter.

Reply:
Yeah, the issue with shared definitions should be done in a separate issue not to be discussed further in this PR. I think having a repository shipping a yaml file containing model-specific features would be a good start.

Reply:
Issue was created https://gitlab.com/stavros/python-yeelight/-/issues/68